master/internal/provisioner/scale_decider.go

```diff
@@ -4,10 +4,17 @@ import (
 	"sort"
 	"time"
 
+	"github.com/determined-ai/determined/master/pkg/actor"
+
 	"github.com/determined-ai/determined/master/internal/scheduler"
 )
 
-const minRetryInterval = 10 * time.Second
+const (
+	minRetryInterval = 10 * time.Second
+	// Allows left-over agents to connect to the master
+	// before terminating them for not having a connection.
+	masterLaunchConnectionBuffer = time.Minute
```

Comment:
blocking: I think we should use `maxStartingPeriod` since this field accounts for both terminating unconnected instances and retrying launching new instances.